exercises/isbn-verifier/isbn_verifier_test.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+
+from isbn_verifier import verify
+
+
+# Tests adapted from `problem-specifications//canonical-data.json` @ v1.0.0
+
+class IsbnVerifierTests(unittest.TestCase):
+
+    def test_valid_isbn(self):
+        self.assertIs(verify('3-598-21508-8'), True)
+
+    def test_invalid_check_digit(self):
+        self.assertIs(verify('3-598-21508-9'), False)
+
+    def test_valid_with_X_check_digit(self):
+        self.assertIs(verify('3-598-21507-X'), True)
+
+    def test_invalid_check_digit_other_than_X(self):
+        self.assertIs(verify('3-598-21507-A'), False)
+
+    def test_invalid_character_in_isbn(self):
+        self.assertIs(verify('3-598-2K507-0'), False)
+
+    def test_invalid_X_other_than_check_digit(self):
+        self.assertIs(verify('3-598-2X507-0'), False)
+
+    def test_valid_isbn_without_separating_dashes(self):
+        self.assertIs(verify('3598215088'), True)
+
+    def test_valid_isbn_without_separating_dashes_with_X_check_digit(self):
+        self.assertIs(verify('359821507X'), True)
+
+    def test_invalid_isbn_without_check_digit_and_dashes(self):
+        self.assertIs(verify('359821507'), False)
+
+    def test_invalid_too_long_isbn_with_no_dashes(self):
+        self.assertIs(verify('3598215078X'), False)
+
+    def test_invalid_isbn_without_check_digit(self):
+        self.assertIs(verify('3-598-21507'), False)
```

Comment:
Here you can use AssertTrue and AssertFalse instead of AssertIs.

I.e.:
```python
    def test_verify_isbn_example_with_X_spaces(self):
        self.assertTrue(isbn_verifier.verify_isbn('955 20 3051 X'))

```

Reply:
`assertFalse` won't fail the test if the tested method returns `None` in comparison with `assertIs`

Consider this:
```python
import unittest

tc = unittest.TestCase()

tc.assertFalse(None) # passes
tc.assertIs(None, True) # fails as expected
```

Do you see the difference, @jamesmcm?

So, I think it's better to use `assertIs` even if readability suffers a bit

Reply:
That's a good point, seems like strange behaviour from the unittest module.

Reply:
Yeah, especially considering that `bool(None)` returns `False` meaning that `None` is Falsy

Reply:
@m-a-ge, your example code is asserting that `None` is Falsey, so passing is the expected behaviour - `bool(None) is False`.

```
import unittest

tc = unittest.TestCase()

tc.assertFalse(None)  # Passes, since None is Falsey
tc.assertTrue(None)   # Fails because None is not Truthy

tc.assertIs(None, False)  # Fails because None is not False
tc.assertIs(None, True)   # Fails because None is not True
```

In theory, there's nothing wrong with using `assertTrue` and `assertFalse` in exercise tests, but the presence of the latter will cause some tests to pass when no code is implemented, and so these could confuse some people. Using `assertIs` creates another slight issue, in that it forces learners to return booleans, even though they may expect to be able to use Truthy and Falsey statements instead. I'd consider `assertIs` the lesser of two evils.

Reply:
@N-Parsons thanks for clarification